### PR TITLE
refactor: Code cleanup for not used code

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

refactor: Code cleanup for not used code

Prepare to merge `new-planner` and `old-planner` to `common-planner`.

### CODE REVIEW

File: src/query/service/src/sql/planner/binder/copy.rs [hunk 1]
```diff
@@ -500,3 +506,123 @@ impl<'a> Binder {
...
+
+/// parse_stage_location_v2 work similar to parse_stage_location.
+///
+/// Difference is input location has already been parsed by parser.
+pub async fn parse_stage_location_v2(
+    ctx: &Arc<dyn TableContext>,
+    name: &str,
+    path: &str,
+) -> Result<(UserStageInfo, String)> {
+    debug_assert!(path.starts_with('/'), "path should starts with '/'");
```
Comment (BohuTANG) [inline src/query/service/src/sql/planner/binder/copy.rs:562]: Why use `debug_assert` here?

Comment (Xuanwo) [reply src/query/service/src/sql/planner/binder/copy.rs:562]: I just copied it from the old place. I plan to refactor them with unit tests.

Comment (Xuanwo) [reply src/query/service/src/sql/planner/binder/copy.rs:562]: Tracked at https://github.com/datafuselabs/databend/issues/7772

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
The next we can remove `settings.get_enable_planner_v2` 🚀

**Pull request merged**